extra_foam/pipeline/f_worker.py

```diff
@@ -152,6 +156,26 @@ def run(self):
                     except Full:
                         pass
 
+                if self._detector_extension is not None and sent:
+                    processed = data_out["processed"]
+                    detector, key = data_out["catalog"].main_detector.split()
+
+                    now = time.time()
+                    bridge_data = {
+                        f"EF_{detector}" : {
+                            key: processed.image.masked_mean,
+                            "metadata": {
+                                "timestamp": now,
+                                "timestamp.sec": int(now),
```

Comment:
I don't know if this is a good idea. The idea was to make it match the time we're finished with processing in extra-foam, but maybe that's dumb since it won't match the timestamp from the train ID.

Reply:
I believe it would be alright. IIRC matching is based on `tid` and not on the timestamp. If it is possible, you could add a matching test on `test_transformer` for a deferred update with a same `tid` and a different timestamp.

Reply:
Decided to remove this in bc79978e, I think it's just too weird to have the other fields of the timestamp not match the train ID.

Reply:
I don't know the exact purpose here, but it is fine that the timestamp does not match the trainId (in karabo at least). in karabo this is metadata information about the pipeline data. by default it takes the time when it is written to an output channel: if you have a data going through multiple devices, then the tid/timestamp are overwritten everytime here.

You should rely on the timestamp information in the data if that is important.

For example in the train matcher, for pipeline data we replicate the source metadata (trainId and timestamp), but for control data, since they are originally distributed via message broker, we don't have this metadata information, so we create a metadata for it when  it is sent out. In this new metadata, the trainId is taken from the source but the timestamp is the one when it is sent out. But the real timestamp (when the property value was updated is contained in the data)

Reply:
Ah hah, interesting :thinking: EXtra-foam doesn't rely on the timestamp, I only put it in there because it's part of the karabo-bridge protocol. But maybe it's better to match what Karabo does, if only for the sake of consistency.

Reply:
Reverted in 0d816ce5.